src/main/java/edu/harvard/iq/dataverse/DataFile.java

```diff
@@ -596,7 +607,10 @@ public FileMetadata getLatestFileMetadata() {
         FileMetadata resultFileMetadata = null;
 
         if (fileMetadatas.size() == 1) {
+            setUnpublished(fileMetadatas.get(0).getDatasetVersion() == null || fileMetadatas.get(0).getDatasetVersion().getVersionState().equals(VersionState.DRAFT));
             return fileMetadatas.get(0);
+        } else {
+            setUnpublished(false); // Since only one can be in Draft assume there is a published version
```

Comment:
Is deaccessioned an issue, or are those files already excluded from the list?

Reply:
FWIW: I wasn't concerned about files in a deaccessioned version showing but what would happen for a file in the draft version when it was only previously in a deaccessioned version. It would have two filemetadatas and thus not show as draft. With the other code changes, I guess the equivalent question is whether deaccessioning the only version(s) a file was published in does/should set it's publication date to null. Probably a minor issue in any case and we should consider changing deaccession behavior here out of scope.